lib/semian.rb

```diff
@@ -150,10 +152,12 @@ def to_s
   #
   # Returns the registered resource.
   def register(name, **options)
-    circuit_breaker = create_circuit_breaker(name, **options)
-    bulkhead = create_bulkhead(name, **options)
-
-    resources[name] = ProtectedResource.new(bulkhead, circuit_breaker)
+    @lock.synchronize do
+      return resources[name] if resources[name]
```

Comment:
The code looks good to me from a safety point of view. 👍 

The issue is that this changes the behaviour of the public API `register`.
Before, future calls to register would overwrite, now they don't.